src/main/webapp/js/login.js

```diff
@@ -0,0 +1,384 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* ==========================================================================
+   USER SIGN-IN
+   ========================================================================== */
+function onSignIn(googleUser) {
+    let id_token = googleUser.getAuthResponse().id_token;
+    let profile = googleUser.getBasicProfile();
+    fetch(`/login?id_token=${id_token}`)
+        .then((response) => response.json())
+        .then((data) => {
+            localStorage.setItem("user", data.sub);
+            localStorage.setItem("loggedIn", true);
+            addUserContent(profile.getName(), profile.getImageUrl());
+            showAccountMenu();
+        }).catch((error) => {
+            console.log(error);
+        });
+}
+
+//Add user information to signed in UI
+function addUserContent(name, image) {
+    // document.getElementById("user-name").innerText = name;
+    document.getElementById("profile-pic").src = image;
+}
+
+//Replaces the sign-in button with signed in UI
+function showAccountMenu() {
+    document.getElementById("account-menu").classList.add("show");
+    document.getElementById("account-menu").classList.remove("hide");
+    document.getElementById("sign-in").classList.add("hide");
+    document.getElementById("sign-in").classList.remove("show");
+}
+
+function hideAccountMenu() {
+    document.getElementById("account-menu").classList.add("hide");
+    document.getElementById("account-menu").classList.remove("show");
+    document.getElementById("sign-in").classList.add("show");
+    document.getElementById("sign-in").classList.remove("hide");
+}
+
+//Logs out of the account
+function signOut() {
+    var auth2 = gapi.auth2.getAuthInstance();
+    auth2.signOut().then(function() {
+        console.log('User signed out.');
```

Comment:
this is unnecessary. Are there any UI changes / logic changes that should be here instead?

Reply:
I don't believe so, I'll remove the promise